### PR TITLE
First instance of transit spectroscopy notebook fitting JWST NIRSpec …

### PR DESCRIPTION
**Pull request by exosing**
…time series data.

 On branch transit_spectroscopy_notebook
 Changes to be committed:
       new file:   3DGrid/mmu_t45g45m00v05.flx
       new file:   Exoplanet_Transmission_Spectra_JWST.ipynb
       new file:   NIRSpec.prism.sensitivity.sav
       new file:   limb_darkening.py
       new file:   notebookrun1/JWST_NIRSpec_Prism_fit_transmission_spectra.csv
       new file:   requirements
       new file:   trans-iso_GJ-0436_0669.0_+0.0_0.56_0001_0.00_model.NIRSpec_PRISM.txt
       new file:   transit_module.py

 Untracked files:
       jwst_data.pickle -> 1 GB data file stored on box drive

### CODE REVIEW

**Comment by robelgeda** [comment]
Hi David, just a heads up: 
1) The data should be uploaded to the box folder I shared with you and load the data into the NB via a box URL/link. Please lmk in slack if you would like instructions on how to do this. 
2) All the code should live inside the notebook or be installed as libraries. So you should define everything in the included `.py` files inside the notebooks or make a small repo on GitHub. This is because we have no way of showing the code that lives in the python script files.
3) All the cell outputs in the notebooks should be cleared before uploading them onto GitHub. This is because the outputs make the commits large significantly contributing to the size of the repository.

**Comment by exosing** [comment]
Data are now uploaded on box.  .py codes are now all contained within the notebook.

**Comment by rosteen** [comment]
@exosing  I'm getting an error when I try to run the notebook after the updates:

`Unreadable Notebook: [...] NotJSONError('Notebook does not appear to be JSON: \'{\\n "cells": [\\n {\\n "cell_type": "m...')`

Would you double check that the copy here matches your running local copy? I'm not sure how this error would have happened, since you wouldn't have merged something that wasn't running on your end.

**Comment by exosing** [comment]
Which version do I check it against?

I did run into some GitHub errors that I tried to fix, though I’m still quite new.

Cheers,
David

**Comment by rosteen** [comment]
If you have time in the next couple days, it would probably be easier to get on a call so we can try to figure this out together rather than go back and forth here (especially since I don't know what's causing this). I'll message you on Slack to set something up.

**Comment by rosteen** [comment]
Actually, scratch that - I found what's causing the error. Your last commit had a merge conflict, so in your notebook there's a couple lines like `<<<<<<< HEAD` that need to be removed. I'll add a suggestion to the file where it needs to be fixed.

File: jdat_notebooks/transit_spectroscopy_notebook/Exoplanet_Transmission_Spectra_JWST.ipynb [hunk 1]
```diff
@@ -1647,6 +1647,7 @@
     "plt.errorbar(wsd,rprs**2*1E6,xerr=werr,yerr=2*rerr*rprs*1E6, fmt='o',zorder=10,color='orange',label='Recovered Spectra')\n",
     "plt.xlabel('Wavelength ($\\mu$m)')\n",
     "plt.ylabel('Transit Depth ($R_p/R_s$)$^2$ (ppm)')\n",
+<<<<<<< HEAD
```
Comment (rosteen) [inline jdat_notebooks/transit_spectroscopy_notebook/Exoplanet_Transmission_Spectra_JWST.ipynb:1650]: This is what's making the notebook unusable. You need to delete either the code between `<<<<<<< HEAD` and `=======` or `=======` and ` >>>>>>> 5e04021b6b55220c191460a6de0d259412d2a6a6`, depending on what you want to keep, and then delete those three lines of git markup. Committing and pushing after fixing that should fix the problem I was seeing.

Comment (exosing) [reply jdat_notebooks/transit_spectroscopy_notebook/Exoplanet_Transmission_Spectra_JWST.ipynb]: Weird, it's clearly some copy paste issue, but I didn't show up at at on my local version, until I tried re-opening it, which failed.

**Comment by rosteen** [comment]
The notebook loads properly for me now. I'm rerunning it and can also confirm that the error I was seeing in the wavelength bin loop is also fixed (I didn't get past bin 0 before, it's not done running but I'm on bin 2 now). 

Looks like your latest commit had some cell output (from the last code cell, I think) included - if you clear the cell output and re-commit I think this is good to go! Thanks for addressing my comments, and for your work in general on this.

**Comment by robelgeda** [comment]
Hi all. I am going to copy over comments I made on the other PR here about the test bot:
> Just checking in, I believe the tests are failing because the some of the requirements in the requirement file are not included in the testing environment file. The testing environment file is at dat_pyinthesky/jdat_notebooks/environment.yml and the new required modules should be listed in there. Please check to avoid duplicates or conflicts as removing or editing modules already in the environment file will cause the test to fail (other notebooks will fail).

**Pull request merged**